x-pack/plugins/reporting/server/export_types/csv_searchsource/generate_csv/generate_csv.ts

```diff
@@ -70,20 +70,35 @@ export class CsvGenerator {
       },
     };
 
-    const results = (
-      await lastValueFrom(this.clients.data.search(searchParams, { strategy: ES_SEARCH_STRATEGY }))
-    ).rawResponse as estypes.SearchResponse<unknown>;
+    let results: estypes.SearchResponse<unknown> | undefined;
+    try {
+      results = (
+        await lastValueFrom(
+          this.clients.data.search(searchParams, { strategy: ES_SEARCH_STRATEGY })
+        )
+      ).rawResponse as estypes.SearchResponse<unknown>;
+    } catch (err) {
+      this.logger.error(`CSV export scan error: ${err}`);
```

Comment:
I find this error does not get caught in time, if the error is a request timeout. 

1. Create a proxy for Elasticsearch that slows down search response data if the request is for scan + scroll
2. Use the proxy to slow down those specific search responses by 30 seconds
3. Note that the Task Manager timeout handler on the report job run function (2 minutes) becomes the error used in the report job failure message.

Two errors are happening: 1 - the request timeout error from this search call (should trigger after 30 seconds?). 2 - the Task Manager task run timeout (should trigger after 2 minutes, or the xpack.reporting.queue.timeout setting).

From the logs, the error I call "2" is actually the primary error captured as the error of the job failure:
```
[2022-08-01T16:30:18.842-07:00][INFO ][plugins.reporting] Scheduled csv_searchsource reporting task. Task ID: task:e78c1020-11f1-11ed-b6fe-4306e3b7e272. Report ID: l6bdwhiz1y4gae92f34z28l8
[2022-08-01T16:30:21.972-07:00][INFO ][plugins.reporting.runTask.csv_searchsource.execute-job:l6bdwhiz1y4gae92f34z28l8] Executing search request...
[2022-08-01T16:30:23.654-07:00][WARN ][plugins.usageCollection.usage-collection.usage-counters-service] ConnectionError: socket hang up - Local: unknown:unknown, Remote: unknown:unknown
    at KibanaTransport.request (/Users/tim/elastic/kibana/node_modules/@elastic/transport/src/Transport.ts:582:17)
[2022-08-01T16:32:21.957-07:00][ERROR][plugins.reporting.runTask] Error: ReportingError(code: unknown_error) "Max attempts (1) reached for job l6bdwhiz1y4gae92f34z28l8. Failed with: Timeout has occurred"
    at mapToReportingError (/Users/tim/elastic/kibana/x-pack/plugins/reporting/common/errors/map_to_reporting_error.ts:50:10)
    at Object.run (/Users/tim/elastic/kibana/x-pack/plugins/reporting/server/lib/tasks/execute_report.ts:433:31)
    at TaskManagerRunner.run (/Users/tim/elastic/kibana/x-pack/plugins/task_manager/server/task_running/task_runner.ts:306:22)
[2022-08-01T16:32:21.987-07:00][ERROR][plugins.reporting.runTask.csv_searchsource.execute-job:l6bdwhiz1y4gae92f34z28l8] CSV export scan error: Error: Request timed out
[2022-08-01T16:32:21.987-07:00][ERROR][plugins.reporting.runTask.csv_searchsource.execute-job:l6bdwhiz1y4gae92f34z28l8] Error: Request timed out
    at getKbnServerError (/Users/tim/elastic/kibana/src/plugins/kibana_utils/server/report_server_error.ts:28:10)
    at search (/Users/tim/elastic/kibana/src/plugins/data/server/search/strategies/es_search/es_search_strategy.ts:55:15)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2022-08-01T16:32:21.988-07:00][WARN ][plugins.reporting.runTask.csv_searchsource.execute-job:l6bdwhiz1y4gae92f34z28l8] No scrollId to clear!
[2022-08-01T16:32:21.988-07:00][INFO ][plugins.reporting.runTask.csv_searchsource.execute-job:l6bdwhiz1y4gae92f34z28l8] Finished generating. Row count: 0.
[2022-08-01T16:32:22.657-07:00][WARN ][plugins.taskManager] Cancelling task report:execute "e78c1020-11f1-11ed-b6fe-4306e3b7e272" as it expired at 2022-08-01T23:32:21.742Z after running for 02m 00s (with timeout set at 120s).
[2022-08-01T16:32:22.658-07:00][WARN ][plugins.reporting.runTask] Cancelling job l6bdwhiz1y4gae92f34z28l8...
```

This means when you view the failed job in Stack Management, you see the error "Max attempts (1) reached for job l6bepriz1zctae92f34qi495. Failed with: Timeout has occurred" (which doesn't say much), instead of "Error: Request timed out" 

<img width="1874" alt="image" src="https://user-images.githubusercontent.com/908371/182266177-d1000b93-4231-480a-b714-e91ce68ada94.png">